File: include/apl/errors/NYIError.java
package APL.errors;

import APL.types.*;

public class NYIError extends APLError { // AKA LazyError
  public NYIError(std::string s) {
    super(s);
  }
  public NYIError(std::string s, Tokenable fun) {
    super(s, fun);
  }
  public NYIError(std::string s, Callable fun, Tokenable cause) {
    super(s, fun, cause);
  }
}

File: APL/errors/NYIError.cpp
#include <APL/APL::errors::NYIError.h>

namespace APL::errors
{
    // using namespace APL::types;

    APL::errors::NYIError::APL::errors::NYIError(std::string const& s) : APLError(s)
    {
    }

    APL::errors::NYIError::APL::errors::NYIError(std::string const& s, std::shared_ptr<APL::types::Tokenable> fun) : APLError(s, fun)
    {
    }

    APL::errors::NYIError::APL::errors::NYIError(std::string const& s, std::shared_ptr<APL::types::Callable> fun, std::shared_ptr<APL::types::Tokenable> cause) : APLError(s, fun, cause)
    {
    }
}
